src/components/App/App.js

```diff
@@ -1,33 +1,56 @@
 import React, { Component } from "react";
 import Header from "../Header/Header.js";
 import Movies from "../Movies/Movies.js";
-import { movieData, selectedMovieDetails } from "../../testData.js";
 import MovieDetails from "../MovieDetails/MovieDetails.js";
 
 class App extends Component {
   constructor() {
     super();
 
     this.state = {
-      movies: movieData.movies,
-      selectedMovie: '',
+      movies: [],
+      selectedMovie: "",
+      error: ""
     };
   }
 
-  showMovieDetails = () => {
-    this.setState({ selectedMovie: selectedMovieDetails.movie });
+  componentDidMount() {
+    fetch("http://rancid-tomatillos.herokuapp.com/api/v2/movies")
+      .then((response) => response.json())
+      .then((movies) => this.setState({ movies: movies.movies, error: "" }))
+      .catch((error) =>
+        this.setState({ error: "Sorry, but an error has been encountered. Please try again later"})
+      );
   }
 
+  showMovieDetails = (movieID) => {
+    fetch(`http://rancid-tomatillos.herokuapp.com/api/v2/movies/${movieID}`)
+      .then((response) => {
+        if (!response.ok) {
+          this.setState({
+            error: "Details for this movie are not available at this time. Please check back later."
+          });
+        } else {
+          return response.json();
+        }
+      })
+      .then((selectedMovie) =>
+        this.setState({ selectedMovie: selectedMovie.movie, error: "" })
+      )
+      .catch((error) => console.log("error"));
```

Comment:
I think this catch is redundant since you've got the check for if the response is ok on line 29